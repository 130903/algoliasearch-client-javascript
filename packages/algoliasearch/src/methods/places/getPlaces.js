// @flow

import { pluralError } from '../../errors';
import type { RequestMethod, ObjectID, RequestOptions } from '../../types';

export default function getPlaces({
  requester,
  objectIDs,
  attributesToRetrieve,
  options,
}: {
  requester: RequestMethod,
  objectIDs: ObjectID[],
  attributesToRetrieve?: string[],
  options?: RequestOptions,
}) {
  if (!Array.isArray(objectIDs)) {
    throw pluralError('getObject');
  }
  if (objectIDs.length === 0) {
    throw new Error('You need to have at least one place to retrieve');
  }
  const attrs = attributesToRetrieve && {
    attributesToRetrieve: attributesToRetrieve.join(','),
  };

  return requester({
    method: 'POST',
    path: '/1/indexes/*/objects',
    body: {
      requests: objectIDs.map(objectID => ({
        indexName: 'places',
        objectID,
        ...attrs,
      })),
    },
    requestType: 'read',
    options,
  });
}
