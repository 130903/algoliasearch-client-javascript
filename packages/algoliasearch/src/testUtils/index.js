// @flow

import type { RequestMethod } from 'algoliasearch-requester';

export const snapshotAll = (requests: RequestMethod[]) =>
  requests.map(req => req.then(sn => expect(sn).toMatchSnapshot()));
