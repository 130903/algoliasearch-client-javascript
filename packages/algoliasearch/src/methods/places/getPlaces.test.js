import { snapshotAll, fakeRequester } from '../../testUtils';
import getPlaces from './getPlaces';

it('regular getPlaces', () => {
  const requests = [
    getPlaces({ requester: fakeRequester, objectIDs: ['some_item', 'bingo'] }),
    getPlaces({
      requester: fakeRequester,
      objectIDs: ['some_item', 'bingo'],
      attributesToRetrieve: ['sing', 'som'],
    }),
  ];
  snapshotAll(requests);
});

it('errors when no objectIDs are given', () => {
  expect(() =>
    getPlaces({ requester: fakeRequester, objectIDs: [] })
  ).toThrow();
});

it('errors when no attributesToRetrieve is not an array', () => {
  expect(() =>
    getPlaces({ requester: fakeRequester, attributesToRetrieve: 'test' })
  ).toThrow();
  expect(() =>
    getPlaces({
      requester: fakeRequester,
      attributesToRetrieve: { cool: 'test' },
    })
  ).toThrow();
});
