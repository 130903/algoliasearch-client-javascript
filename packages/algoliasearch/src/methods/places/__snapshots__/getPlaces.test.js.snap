// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regular getPlaces 1`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "indexName": "places",
        "objectID": "some_item",
      },
      Object {
        "indexName": "places",
        "objectID": "bingo",
      },
    ],
  },
  "method": "POST",
  "options": undefined,
  "path": "/1/indexes/*/objects",
  "requestType": "read",
}
`;

exports[`regular getPlaces 2`] = `
Object {
  "body": Object {
    "requests": Array [
      Object {
        "attributesToRetrieve": "sing,som",
        "indexName": "places",
        "objectID": "some_item",
      },
      Object {
        "attributesToRetrieve": "sing,som",
        "indexName": "places",
        "objectID": "bingo",
      },
    ],
  },
  "method": "POST",
  "options": undefined,
  "path": "/1/indexes/*/objects",
  "requestType": "read",
}
`;
